controllers/isReviewedController.js

```diff
@@ -0,0 +1,42 @@
+const db = require('../models/isReviewedModel');
+
+// Defining methods for the postsController
+module.exports = {
+  findAll: function (req, res) {
+    db.find(req.query)
```

Comment:
**this is pretty a serious bug**

```javascript
db.find(req.query) // Not Compliant
```

User provided data such as URL parameters, POST body-content should always be considered untrusted and tainted. Applications performing NoSQL operations based on tainted data could be exploited in a way similar to SQL Injection, where an attacker could inject NoSQL objects to access sensitive information or compromise data integrity.

To fix this and have a _compliant solution_ the data should be sanitized like so:

**example**

```javascript
function (req, res) {
  let query = { user: req.query.user.toString(), city: req.query.city.toString() };

  db.collection("users")
    .find(query) // Compliant
    .toArray((err, docs) => { });
}
```

Reply:
I added a fix by setting the query to a "toString()" method.

`findAll: function (req, res) {
    db.find(req.query.toString())`

Reply:
implemented a fix 

  create: function (req, res) {
    let reviewData = new db({
      name: req.body.name,
      imageUrl: req.body.imageUrl,
      imageKey: req.body.imageKey,
      description: req.body.description,
      totalStars: req.body.totalStars,
      reviewBody: req.body.reviewBody,
      isActive: req.body.isActive,
      isReviewable: req.body.isReviewable,
      created: req.body.created,
      updated: req.body.updated,
    });

    db.create(reviewData)
      .then((dbModel) => res.json(dbModel))
      .catch((err) => res.status(422).json(err));
  },